binr/rabin2/rabin2.c

```diff
@@ -112,25 +114,35 @@ static bool isBinopHelp(const char *op) {
 	return false;
 }
 
-static bool extract_binobj (const RBinFile *bf, const RBinObject *o, int idx) {
-	ut64 boffset = o ? o->boffset : 0;
-	ut64 bin_size = o ? o->obj_size : 0;
-	const ut8 *bytes;
+static bool extract_binobj (const RBinFile *bf, const RBinXtrData *data, int idx) {
+	ut64 boffset = data ? data->offset : 0;
+	ut64 bin_size = data ? data->size : 0;
+	char *bytes;
+	char *bytes_encoded;
 	//ut64 sz = bf ? r_buf_size (bf->buf) : 0;
-	RBinInfo *info = o ? o->info : NULL;
-	const char *arch = info ? info->arch : "unknown";
-	char bits = info ? info->bits : 0;
+	char *arch = "unknown";
+	int bits = 0;
+	char *libname = NULL;
 	const char *filename = bf ? bf->file : NULL;
 	char *path = NULL, *outpath = NULL, *outfile = NULL, *ptr = NULL;
 	ut32 outfile_sz = 0, outpath_sz = 0;
 	bool res = false;
 
-	if (!bf || !o || !filename ) return false;
+	if (!bf || !data || !filename ) return false;
+	if (data->metadata) {
+		arch = data->metadata->arch;
+		bits = data->metadata->bits;
+		libname = data->metadata->libname;
+	}
 	if (bin_size == bf->size && bin_size) {
 		eprintf ("This is not a fat bin\n");
 		return false;
 	}
-	bytes = r_buf_buffer (bf->buf);
+	ut64 size = 0;
+	bytes_encoded = sdb_get (data->sdb, sdb_fmt (0, "%d", data->offset), 0);
+	bytes = sdb_decode (bytes_encoded, &size);
```

Comment:
use data->size since RBinXtrData holds that info so `sdb_decode (bytes_encode, NULL);` this is causing bad parsing i was getting `cmdsize < 1` in mach parser.  sdb_decode afaik it returns in that len parametrs the len of the encoding that differs from the bytes decoded